memcache/stats.go

```diff
@@ -0,0 +1,474 @@
+/*
+Copyright 2011 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http:// www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package memcache
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type errorsSlice []error
+
+func (errs errorsSlice) Error() string {
+	switch len(errs) {
+	case 0:
+		return ""
+	case 1:
+		return errs[0].Error()
+	}
+	n := (len(errs) - 1)
+	for i := 0; i < len(errs); i++ {
+		n += len(errs[i].Error())
+	}
+
+	var b strings.Builder
+	b.Grow(n)
+	b.WriteString(errs[0].Error())
+	for _, err := range errs[1:] {
+		b.WriteByte(':')
+		b.WriteString(err.Error())
+	}
+	return b.String()
+}
+
+func (errs *errorsSlice) AppendError(err error) {
+	if err == nil {
+		return
+	}
+	if errs == nil {
+		errs = &errorsSlice{err}
+		return
+	}
+	set := *errs
+	set = append(set, err)
+	*errs = set
+}
+
+// ServerStats contains the general statistics from one server.
+type ServerStats struct {
+	//ServerErr Error if can't get the stats information
```

Comment:
Another space after //

Also, these should all be complete sentences (https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences) ending in periods, correct capitalization, etc.